lib/AnalysisStructured/PtrAnalysis.cpp

```diff
@@ -99,14 +205,47 @@ LogicalResult PtrState::addState(const PtrState &lhsState,
     scalar = lhsState.scalar ? lhsState.scalar : rhsState.scalar;
   }
 
-  for (uint64_t i = 0; i < lhsState.getRank(); i++) {
-    auto newOffset =
-        addOFRs(lhsState.offsets[i], rhsState.offsets[i], loc, builder);
-    offsets.push_back(newOffset);
+  if (!lhsState.isStructured() && !rhsState.isStructured()) {
+    if (lhsState.getNonContinuousDim() != rhsState.getNonContinuousDim()) {
+      op->emitRemark("PtrAnalysis: do not support adding two pointer states "
+                     "that have different non-continuous dimension");
+      return failure();
+    }
+  }
 
-    auto newStride =
-        addOFRs(lhsState.strides[i], rhsState.strides[i], loc, builder);
-    strides.push_back(newStride);
+  for (uint64_t i = 0; i < lhsState.getRank(); i++) {
+    if (!lhsState.dimIsNotContinuous(i) && !rhsState.dimIsNotContinuous(i)) {
+      auto newOffset =
```

Comment:
nit: I find having negative in function names make it hard to read, specifically for case like these where you end up doing `!notSomething()`. Can you change all of the methods to check for continuous instead?